packages/core/src/Cardano/types/Block.ts

```diff
@@ -74,18 +77,42 @@ export const SlotLeader = (value: string): SlotLeader => {
   }
 };
 
-export interface Block {
+/**
+ * Get Bech32 encoded VRF verification key from base64 encoded string
+ *
+ * @param value is a Base64 string
+ * @returns Bech32 encoded vrf_vk
+ */
+export const VrfVkBech32FromBase64 = (value: string) =>
+  VrfVkBech32(CSL.VRFVKey.from_bytes(Buffer.from(value, 'base64')).to_bech32('vrf_vk'));
+
+/** Minimal Block type that can be  */
+// REVIEW: made some of the properties `optional` because some may be undefined if block is Byron
```

Comment:
Let's hoist the optional properties to `Block` type as non-optional (as they originally were).

We can address those in a separate task: compute`size` with CSL. For `fee` we could try sum of `CSL.Transaction.fromHex(block.body.txPayload[n].raw).body().fee()`

Reply:
A solution for the optional properties is here: [1fad239](https://github.com/input-output-hk/cardano-js-sdk/pull/450/commits/1fad2392e4bb48c94364c6fc6d9f7e618e6533ad)